cmd/soroban-rpc/internal/db/ledgerentry.go

```diff
@@ -260,16 +260,22 @@ func entryKeyToTTLEntryKey(key xdr.LedgerKey) (xdr.LedgerKey, error) {
 }
 
 func (l *ledgerEntryReadTx) GetLedgerEntries(keys ...xdr.LedgerKey) ([]LedgerKeyAndEntry, error) {
-	encodedKeys := make([]string, len(keys), 2*len(keys))
-	encodedKeyToKey := make(map[string]xdr.LedgerKey, len(keys))
-	encodedKeyToEncodedTTLLedgerKey := make(map[string]string, len(keys))
-	for _, k := range keys {
+	encodedKeys := make([]string, 0, 2*len(keys))
+	type keyToEncoded struct {
+		key           xdr.LedgerKey
+		encodedKey    string
+		encodedTTLKey *string
+	}
```

Comment:
Can you explain why one is a pointer while the other isn't? Is it because not every ledger key has a corresponding TTL key (e.g. an account entry)? I thought this was the explanation but I can't find the part where we skip over non-expiring entries in the code.

Reply:
>  Is it because not every ledger key has a corresponding TTL key (e.g. an account entry)? 

Exactly

They are skipped at:

```go
		if !hasTTLKey(k) {
			continue
		}
```